docs/dev/plugins.md

```diff
@@ -148,14 +175,95 @@ If things go right, the Status now has a bunch of Activators, each with a block
+
+After activation, code for that aspect of the plugin is loaded and ready to execute. Executioon may be triggered by a number of different events. For example, the CliCommand plugin types are implicitly executed by Thor when `Inspec::CLI` calls `start()`.
+
+Refer to the sections below for details about activation and execution timing.
+
+## Implementing a CLI Command Plugin
+
+The CliCommand plugin_type allows you to extend the InSpec command line interface by adding a namespace of new commands. InSpec is based on [Thor](http://whatisthor.com/) ([docs](https://www.rubydoc.info/github/wycats/thor/Thor)), and the plugin system exposes Thor directly.
+
+CliCommand can do things like:
+
+```bash
+# A namespaced custom command with options
+you@machine$ inspec sweeten add --kind sugar --teaspoons 2
+# A namespaced custom command with short options
+you@machine$ inspec sweeten add -k agave
+# Mix global and namespace options
+you@machine$ inspec --debug sweeten add -k aspartame
+# Namespace included in help
+you@machine$ inspec help
+Commands:
+  inspec archive PATH      # archive a profile to tar.gz (default) or zip
+  inspec sweeten ...       # Add spoonfuls til the medicine goes down
+# Detailed help
+[cwolfe@lodi inspec-plugins]$ be inspec help sweeten
+Commands:
+  inspec sweeten add [opts]       # Adds sweetener to your beverage
+  inspec sweeten count            # Reports on teaspoons in your beverage, always bad news
+```
+
+Currently, it cannot do a direct (non-namespaced) command
+```bash
+
+
+
+### Declare your plugin activators
+
+In your `plugin.rb`, include one or more `cli_command` activation blocks.  The activation block name will be matched against the command line arguments; if the name is present, your activator will fire (in which case it should load any needed libraries) and should return your implementation class.
+
+#### CliCommand Activator Example
+
+```ruby
+
+# In plugin.rb
+module InspecPlugins::Sweeten
+  class Plugin < Inspec.plugin(2)
+    # ... other plugin stuff
+
+    cli_command :sweeten do
+      require_relative 'cli.rb'
+      InspecPlugins::Sweeten::CliCommand
+    end
+  end
+end
+```
+
+Like any activator, the block above will only be called if needed. For CliCommand plugins, the plugin system naively scans through ARGV, looking for the activation name as a whole element.  Multiple CliCommand activations may occur if several different names match, though each activation will only occur once.
+
+```bash
+you@machine $ inspec sweeten ... # Your CliCommand implementation is activated and executed
+you@machine $ inspec exec ... # Your CliCommand implementation is not activated
+```
+
+Execution occurs implicitly via `Thor.start()`, which is handled by `bin/inspec`. Keep reading.
+
+### Implementation class for CLI Commands
+
+In your `cli.rb`, you should begin by requesting the superclass from Inspec:
```

Comment:
Should we change this to "from `Inspec.plugin`"? My OO knowledge is not the greatest.